devito/profiling.py

```diff
@@ -137,17 +137,25 @@ def summary(self, dim_sizes, dtype):
             time = self.timings[profile.name]
 
             # Flops
-            itershape = [i.extent(finish=dim_sizes.get(dims[i].name)) for i in itspace]
+            itershape = []
+            for i in itspace:
+                if dims[i].is_Fixed:
+                    start = 0
+                    end = dims[i].size
+                else:
+                    start = arguments[dims[i].start_name]
+                    end = arguments[dims[i].end_name]
+                itershape.append(i.extent(finish=end, start=start))
             iterspace = reduce(operator.mul, itershape)
             flops = float(profile.ops*iterspace)
             gflops = flops/10**9
 
             # Compulsory traffic
             datashape = [i.dim.size if i.dim.is_Fixed
```

Comment:
perhaps this is movable inside the ``for i in itspace`` loop above?

Reply:
TBH I don't understand the reason for calculating the datashape and iterspace separately. They seem to resolve to the same number always. Either that or my changes to the code in this section are based on incorrect assumptions.

Reply:
the itershape differs in general because of the stencil

Reply:
The itershape is the product of the size of all dimensions, and so is the datashape. Correct? Always?

Reply:
no it's the product of the extents. Previously ``dim_sizes`` was being used, which in turn was used to call ``extents``

Reply:
Which one? Itershape or datashape? What is the other one?

Reply:
itershape

Reply:
And how is datashape different?

Reply:
You're touching boundary data through the stencil

Reply:
Pls add a comment

Reply:
reminder that this still needs to be addressed.

Also, since you're touching the profiling engine, can you verify that, for example, acoustic or tti still produce the same "performance summary" when running from ``master`` and from this ``branch`` ? Can you paste here some data?

Reply:
reminder. This still needs to be addressed.

Reply:
Let me pick this apart. @FabioLuporini are you saying that the original code is wrong? Or is this functionally the same as the previous code but you have a different feeling for how it should be styled? If so, can we distinguish between advisory and required changes and style changes can potentially be put into a separate PR and applied consistently. Essentially (and particularly for large PR's) we don't want to have moving goalposts.

Reply:
I originally pointed out something about style (the ``dataspace`` calculation). Then I thought that even more important would be to verify that the "Performance Summary" is correct. For example, if we run ``acoustic`` on ``master`` and then on this branch, does Devito produce/display the same numbers (itershape, datashape, ..)? In the past, @mloubout and I had to do some serious investigation to find out why the performance numbers were off, so I just wanna make sure that we're still producing exactly the same reports as ``master`` (hence my "reminder" from a few days ago).

BTW, the style change was a one-line edit. I'm not such a great believer in "this low-priority thing will be improved in a subsequent PR", because it rarely happens in practice. So it's much better to do it straight away. This is the philosophy we've adopted since the Devito3.0 rewrite (Mathias can confirm)